app/screens/home/search/results/file_options.tsx

```diff
@@ -0,0 +1,227 @@
+    const isTablet = useIsTablet();
+    const serverUrl = useServerUrl();
+    const [action, setAction] = useState<GalleryAction>('none');
+
+    const galleryItem = {...fileInfo, type: 'image'} as GalleryItemType;
+
+    const handleDownload = useCallback(async () => {
+        setAction('downloading');
+    }, []);
+
+    const handleCopyLink = useCallback(async () => {
+        setAction('copying');
+        dismissBottomSheet();
+    }, []);
+
+    const handleGotoChannel = useCallback(async () => {
+        const fetchedPost = await fetchPostById(serverUrl, fileInfo.post_id, true);
+        const post = fetchedPost.post;
+        const rootId = post?.root_id || post?.id;
+        if (rootId) {
+            fetchAndSwitchToThread(serverUrl, rootId);
+        } else {
+            // what to do?
+        }
+    }, [fileInfo]);
+
+    const onDownloadSuccess = () => {
+        dismissBottomSheet();
+    };
+
+    const handlePress = (item: FileOption) => {
+        switch (item.type) {
+            case OptionsActions.DOWNLOAD:
+                handleDownload();
+                break;
+            case OptionsActions.GOTO_CHANNEL:
+                handleGotoChannel();
+                break;
+            case OptionsActions.COPY_LINK:
+                handleCopyLink();
+                break;
+            default:
+        }
+    };
+
+    const renderLabelComponent = useCallback((item: FileOption) => {
+        return (
+            <FormattedText
+                style={style.menuText}
+                id={item.id}
+                defaultMessage={item.defaultMessage}
+            />
+        );
+    }, [style]);
+
+    const renderHeader = () => {
+        const size = getFormattedFileSize(fileInfo.size);
+        return (
+            <View style={style.headerContainer}>
+                <View style={style.iconContainer}>
+                    <FileIcon
+                        file={fileInfo}
+                        iconSize={72}
+                    />
+                </View>
+                <Text
+                    style={style.nameText}
+                    numberOfLines={2}
+                    ellipsizeMode={'tail'}
+                >
+                    {fileInfo.name}
+                </Text>
+                <View style={style.infoContainer}>
+                    <Text style={style.infoText}>{`${size} • `}</Text>
+                    <FormattedDate
+                        style={style.date}
+                        format={format}
+                        value={fileInfo.create_at as number}
+                    />
+                </View>
+            </View>
+        );
+    };
+
+    const renderItem = useCallback(({item}: {item: FileOption}) => {
+        return (
+            <MenuItem
+                labelComponent={renderLabelComponent(item)}
+                iconName={item.iconName}
+                iconContainerStyle={style.selected}
+                onPress={() => handlePress(item)}
```

Comment:
There are several issues here.
- You are creating a new function on every render.
- `handlePress`  depends on `handleGoToChannel`, which also depends on fileInfo. But `renderItem` does not change in case `fileInfo` changes, so technically this may return a wrong result. That being said, `fileInfo` seems "constant" on this context, so may not be such a big deal.

0/5 if this needs any change. You can leave it as it is for now, and we can check the performance implications later.

Reply:
Perhaps I should just use 3 instances of `MenuItem` and get rid of the flatlist?

Reply:
I'm looking through the Gallery implementation and realize there are other items that need to be accounted for and I can possibly use some of those components.

Reply:
I removed the FlatList implementation all together and render each OptionItem seperately now.